<robot name="gray_chair">
    <link name="base_link_gc">
        <visual>
            <geometry>
                <cylinder length="0.47" radius="0.01"/>
            </geometry>
            <origin rpy="1.57079633 0 0" xyz="0.2 0 0.44"/>
            <material name="silver">
                <color rgba="0.5 0.5 0.5 1"/>
            </material>
        </visual>
    </link>
    
    <link name="base_link_gc1">
        <visual>
            <geometry>
                <cylinder length="0.4" radius="0.01"/>
            </geometry>
            <origin rpy="0 1.57079633 0" xyz="0 0.235 0.44"/>
            <material name="silver"/>
        </visual>
    </link>
    
    <link name="base_link_gc2">
        <visual>
            <geometry>
                <cylinder length="0.4" radius="0.01"/>
            </geometry>
            <origin rpy="0 1.57079633 0" xyz="0 -0.235 0.44"/>
            <material name="silver"/>
        </visual>
    </link>

    <link name="base_link_gc3">
        <visual>
            <geometry>
                <cylinder length="0.47" radius="0.01"/>
            </geometry>
            <origin rpy="1.57079633 0 0" xyz="-0.2 0 0.44"/>
            <material name="silver"/>
        </visual>
    </link>

    <link name="base_link_gc12">
        <visual>
            <geometry>
                <cylinder length="0.24" radius="0.01"/>
            </geometry>
            <origin rpy="0 1.57079633 0" xyz="0.03 0.235 0.65"/>
            <material name="silver"/>
        </visual>
    </link>
    
    <link name="base_link_gc22">
        <visual>
            <geometry>
                <cylinder length="0.24" radius="0.01"/>
            </geometry>
            <origin rpy="0 1.57079633 0" xyz="0.03 -0.235 0.65"/>
            <material name="silver"/>
        </visual>
    </link>

    <link name="base_link_gc13">
        <visual>
            <geometry>
                <box size="0.15 0.05 0.02"/>
            </geometry>
            <origin rpy="0 0 0" xyz="-0.02 0.235 0.65"/>
            <material name="gray_4">
                <color rgba="0.67 0.65 0.64 1"/>
            </material>
        </visual>
    </link>
    
    <link name="base_link_gc23">
        <visual>
            <geometry>
                <box size="0.15 0.05 0.02"/>
            </geometry>
            <origin rpy="0 0 0" xyz="-0.02 -0.235 0.65"/>
            <material name="gray_4"/>
        </visual>
    </link>

    <link name="seat_gc">
        <visual>
            <geometry>
                <box size="0.4 0.45 0.02"/>
            </geometry>
            <origin rpy="0 0 0" xyz="-0.02 0 0.45"/>
            <material name="beige">
                <color rgba="1 0.9 0.8 1"/>
            </material>
        </visual>
    </link>

    <link name="back_gc">
        <visual>
            <geometry>
                <box size="0.3 0.47 0.02"/>
            </geometry>
            <origin rpy="0 1.7 0" xyz="0.23 0 0.68"/>
            <material name="gray_4"/>
        </visual>
    </link>

    <link name="back_gc_2">
        <visual>
            <geometry>
                <box size="0.1 0.06 0.02"/>
            </geometry>
            <origin rpy="0 1.7 0" xyz="0.205 -0.205 0.49"/>
            <material name="gray_4"/>
        </visual>
    </link>

    <link name="back_gc_3">
        <visual>
            <geometry>
                <box size="0.1 0.06 0.02"/>
            </geometry>
            <origin rpy="0 1.7 0" xyz="0.205 0.205 0.49"/>
            <material name="gray_4"/>
        </visual>
    </link>
    
    <joint name="base_to_base_1_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="base_link_gc1"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>
    
    <joint name="base_to_base_2_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="base_link_gc2"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="base_1_to_base_3_gc" type="fixed">
        <parent link="base_link_gc1"/>
        <child link="base_link_gc3"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="base_to_base_12_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="base_link_gc12"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>
    
    <joint name="base_to_base_22_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="base_link_gc22"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>
    
    <joint name="base_to_base_13_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="base_link_gc13"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>
    
    <joint name="base_to_base_23_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="base_link_gc23"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="base_to_seat_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="seat_gc"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="seat_to_back_gc" type="fixed">
        <parent link="seat_gc"/>
        <child link="back_gc"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="seat_to_back_gc_2" type="fixed">
        <parent link="seat_gc"/>
        <child link="back_gc_2"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="seat_to_back_gc_3" type="fixed">
        <parent link="seat_gc"/>
        <child link="back_gc_3"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <link name="leg_0_gc">
        <visual>
            <geometry>
		<cylinder length="0.4" radius="0.01"/>
            </geometry>
            <origin rpy="0 0 0" xyz="-0.2 0.235 0.23"/>
	    <material name="silver"/>
        </visual>
    </link>

    <link name="wheel_0_gc">
        <visual>
	    <geometry>
                <sphere radius="0.03"/>
            </geometry>
            <origin rpy="0 0 0" xyz="-0.2 0.235 0.015"/>
            <material name="grey3">
                <color rgba="0.1 0.1 0.1 1"/>
            </material>
        </visual>
    </link>

    <joint name="base_to_leg_0_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="leg_0_gc"/>
        <origin xyz="0 0 0"/>
        <axis xyz="0 0 0"/>
    </joint>

    <joint name="leg_0_gc_to_wheel_0_gc" type="fixed">
        <parent link="leg_0_gc"/>
        <child link="wheel_0_gc"/>     
        <origin xyz="0 0 0"/>
        <axis xyz="0 0 0"/>
    </joint>
    
    <link name="leg_1_gc">
        <visual>
            <geometry>
                <cylinder length="0.4" radius="0.01"/>
            </geometry>
            <origin rpy="0 0 0" xyz="-0.2 -0.235 0.23"/>
            <material name="silver"/>
        </visual>
    </link>

    <link name="wheel_1_gc">
        <visual>
            <geometry>
                <sphere radius="0.03"/>
            </geometry>
            <origin rpy="0 0 0" xyz="-0.2 -0.235 0.015"/>
            <material name="grey3"/>
        </visual>
    </link>

    <joint name="base_to_leg_1_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="leg_1_gc"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="leg_1_gc_to_wheel_1_gc" type="fixed">
        <parent link="leg_1_gc"/>
        <child link="wheel_1_gc"/>     
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>
    
    <link name="leg_2_gc">
        <visual>
            <geometry>
                <cylinder length="0.62" radius="0.01"/>
            </geometry>
            <origin rpy="0 -0.2 0" xyz="0.21 0.235 0.34"/>
            <material name="silver"/>
        </visual>
    </link>

    <link name="wheel_2_gc">
        <visual>
            <geometry>
                <sphere radius="0.03"/>
            </geometry>
            <origin rpy="1.5707963 0 0" xyz="0.27 0.235 0.015"/>
            <material name="grey3"/>
        </visual>
    </link>

    <joint name="base_to_leg_2_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="leg_2_gc"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="leg_2_gc_to_wheel_2_gc" type="fixed">
        <parent link="leg_2_gc"/>
        <child link="wheel_2_gc"/>     
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>
    
    <link name="leg_3_gc">
        <visual>
            <geometry>
                <cylinder length="0.62" radius="0.01"/>
            </geometry>
            <origin rpy="0 -0.2 0" xyz="0.21 -0.235 0.34"/>
            <material name="silver"/>
        </visual>
    </link>

    <link name="wheel_3_gc">
        <visual>
            <geometry>
                <sphere radius="0.03"/>
            </geometry>
            <origin rpy="0 0 0" xyz="0.27 -0.235 0.015"/>
            <material name="grey3"/>
        </visual>
    </link>

    <joint name="base_to_leg_3_gc" type="fixed">
        <parent link="base_link_gc"/>
        <child link="leg_3_gc"/>
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>

    <joint name="leg_3_gc_to_wheel_3_gc" type="fixed">
        <parent link="leg_3_gc"/>
        <child link="wheel_3_gc"/>     
        <origin xyz="0 0 0"/>
        <axis xyz="1 0 0"/>
    </joint>
    
    
  </robot>
